client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx

```diff
@@ -2,34 +2,32 @@ import { Button, Gridicon } from '@automattic/components';
 import classnames from 'classnames';
 import { translate, useRtl } from 'i18n-calypso';
 import Badge from 'calypso/components/badge';
-import { isTaskDisabled, hasIncompleteDependencies } from './task-helper';
 import { Task } from './types';
 
 const ChecklistItem = ( { task }: { task: Task } ) => {
 	const isRtl = useRtl();
-	const { id, isCompleted, keepActive, title, actionDispatch } = task;
-	const taskDisabled = isTaskDisabled( task );
+	const { id, completed, disabled, title, actionDispatch } = task;
 
 	// Display chevron if task is incomplete. Don't display chevron and badge at the same time.
-	const shouldDisplayChevron =
-		! hasIncompleteDependencies( task ) && ! isCompleted && ! task.badgeText;
+	const shouldDisplayChevron = ! completed && ! disabled && ! task.badgeText;
 
 	return (
 		<li
 			className={ classnames( 'launchpad__task', {
-				'completed-and-active': isCompleted && keepActive, // a task that is completed and can be revisited
-				'completed-and-inactive': isCompleted && ! keepActive, // a task that is completed and can't be revisited
-				'not-completed': ! isCompleted && ! keepActive, // a task that hasn't been completed yet
+				completed: completed,
+				pending: ! completed,
+				enabled: ! disabled,
+				disabled: disabled,
 			} ) }
 		>
 			<Button
 				className="launchpad__checklist-item"
-				disabled={ taskDisabled }
+				disabled={ disabled }
```

Comment:
now the built-in `disabled` task prop integrates nicely with the actual disabled attribute of the HTML element (single source of truth) where previously we had the `keepActive` prop which was another abstraction